plugins/woocommerce-admin/client/wp-admin-scripts/command-palette/use-add-tracking-to-external-commands.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+import { dispatch, useSelect } from '@wordpress/data';
+import { recordEvent } from '@woocommerce/tracks';
+import { store as commandsStore } from '@wordpress/commands';
+
+// Hook to add tracking to non-WooCommerce commands.
+export const useAddTrackingToExternalCommands = ( origin ) => {
+	const originRef = useRef( origin );
+	const commandsWithTracking = useRef( [] );
+	const commandLoadersWithTracking = useRef( [] );
+	const { commands, commandLoaders, isCommandPaletteOpen } = useSelect(
+		( select ) => {
+			const { getCommands, getCommandLoaders, isOpen } =
+				select( commandsStore );
+			return {
+				commands: getCommands(),
+				commandLoaders: getCommandLoaders(),
+				isCommandPaletteOpen: isOpen(),
+			};
+		},
+		[]
+	);
+
+	useEffect( () => {
+		originRef.current = origin;
+	}, [ origin ] );
+
+	useEffect( () => {
+		if ( ! isCommandPaletteOpen ) {
+			commandsWithTracking.current = [];
+			commandLoadersWithTracking.current = [];
+		}
+	}, [ isCommandPaletteOpen ] );
+
+	useEffect( () => {
+		commands.forEach( ( command ) => {
+			if ( ! commandsWithTracking.current.includes( command.name ) ) {
```

Comment:
Another nitpick: what do you think of using a `Set` here to check for the presence of the augmented command? The array of commands is unlikely to grow to levels in which we could perceive the difference… but since we are changing stuff on the site editor and adding tracking… I sorta feel like everything counts a bit. Also, I don't see any downsides, so perhaps worth it?